hatchet/readers/timemory_reader.py

```diff
@@ -236,13 +237,14 @@ def parse_node(_key, _dict, _hparent, _rank):
             _rank_dict = _keys if self.per_rank else _extra
 
             # handle the rank
-            _rank_dict["rank"] = collapse_ids(_rank)
+            _rank_dict["rank"] = int(collapse_ids(_rank))
             if _rank_dict["rank"] is None:
                 del _rank_dict["rank"]
                 self.include_nid = False
 
             # extract some relevant data
-            _tid_dict["tid"] = collapse_ids(_dict["node"]["tid"])
+            # TODO: Error when there are more than 1 tids
+            _tid_dict["thread"] = collapse_ids(_dict["node"]["tid"])
             _extra["pid"] = collapse_ids(_dict["node"]["pid"])
```

Comment:
```suggestion
            _extra["pid"] = collapse_ids(_dict["node"]["pid"], False)
```

should always be false